analyzers/tests/SonarAnalyzer.UnitTest/Rules/SymbolicExecution/RestrictDeserializedTypesTest.cs

```diff
@@ -18,69 +18,116 @@
+        .WithBasePath(@"SymbolicExecution\Sonar")
+        .AddReferences(AdditionalReferences())
+        .WithOnlyDiagnostics(RestrictDeserializedTypes.S5773);
+
+    private readonly VerifierBuilder roslynCS = new VerifierBuilder()
+        .AddAnalyzer(() => new CS.SymbolicExecutionRunner(AnalyzerConfiguration.AlwaysEnabled))
+        .WithBasePath(@"SymbolicExecution\Roslyn")
+        .AddReferences(AdditionalReferences())
+        .WithOnlyDiagnostics(ChecksCS.RestrictDeserializedTypes.S5773);
 
 #if NETFRAMEWORK // These serializers are available only when targeting .Net Framework
 
-        [TestMethod]
-        public void RestrictDeserializedTypesFormatters_CSharp8()
-        {
-            using var _ = new AssertIgnoreScope(); // EnsureStackState fails an assertion in this test file
-            sonar.AddPaths("RestrictDeserializedTypes.cs")
-                .WithOptions(ParseOptionsHelper.FromCSharp8)
-                .Verify();
-        }
-
-        [TestMethod]
-        public void RestrictDeserializedTypes_DoesNotRaiseIssuesForTestProject_CSharp8() =>
-            sonar.AddPaths("RestrictDeserializedTypes.cs")
-                .WithOptions(ParseOptionsHelper.FromCSharp8)
-                .AddTestReference()
-                .VerifyNoIssueReported();
-
-        [TestMethod]
-        public void RestrictDeserializedTypesJavaScriptSerializer_CSharp8() =>
-            sonar.AddPaths("RestrictDeserializedTypes.JavaScriptSerializer.cs")
-                .WithOptions(ParseOptionsHelper.FromCSharp8)
-                .Verify();
-
-        [TestMethod]
-        public void RestrictDeserializedTypesLosFormatter_CSharp8() =>
-            sonar.AddPaths("RestrictDeserializedTypes.LosFormatter.cs")
-                .WithOptions(ParseOptionsHelper.FromCSharp8)
-                .Verify();
-
-        private static IEnumerable<MetadataReference> AdditionalReferences() =>
-            FrameworkMetadataReference.SystemRuntimeSerialization
-            .Union(FrameworkMetadataReference.SystemRuntimeSerializationFormattersSoap)
-            .Union(FrameworkMetadataReference.SystemWeb)
-            .Union(FrameworkMetadataReference.SystemWebExtensions);
+    [TestMethod]
+    public void RestrictDeserializedTypesFormatters_Sonar_CSharp8()
+    {
+        using var _ = new AssertIgnoreScope(); // EnsureStackState fails an assertion in this test file
+        sonar.AddPaths("RestrictDeserializedTypes.cs")
+            .WithOptions(ParseOptionsHelper.FromCSharp8)
+            .Verify();
+    }
+
+    [Ignore] // ToDo: Remove after implementation
+    [TestMethod]
+    public void RestrictDeserializedTypesFormatters_Roslyn_CSharp8()
+    {
+        using var _ = new AssertIgnoreScope(); // EnsureStackState fails an assertion in this test file
+        roslynCS.AddPaths("RestrictDeserializedTypes.cs")
+            .WithOptions(ParseOptionsHelper.FromCSharp8)
```

Comment:
Split it into two test files:
- **RestrictDeserializedTypes.cs**: test cases which need only C# 7.3
- **RestrictDeserializedTypes.CSharp8.cs**: test cases which use C# 8.0 language features (like coalescing assignment)